pylint/checkers/ellipsis_checker.py

```diff
@@ -0,0 +1,41 @@
+"""Ellipsis checker for Python code
+"""
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+class EllipsisChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "unnecessary_ellipsis"
+    msgs = {
+        "W2301": (
+            "Unnecessary ellipsis constant",
+            "unnecessary-ellipsis",
+            "Used when the ellipsis constant is encountered and can be avoided.",
```

Comment:
Let's add an explanation as to why these are unnecessary. New python users might not know why they are and this message description doesn't necessarily help them avoid future warnings.